src/market/market_state.py

```diff
@@ -0,0 +1,130 @@
+import math
+from typing import NamedTuple
+
+from solana.publickey import PublicKey
+from solana.rpc.api import Client
+
+from src._layouts.market import MINT_LAYOUT
+from src.utils import load_bytes_data
+
+
+class AccountFlags(NamedTuple):
+    initialized: bool = False
+    market: bool = False
+    open_orders: bool = False
+    request_queue: bool = False
+    event_queue: bool = False
+    bids: bool = False
+    asks: bool = False
+
+
+class MarketState(NamedTuple):
+    account_flags: AccountFlags = AccountFlags()
+    own_address: PublicKey = PublicKey(0)
+    vault_signer_nonce: int = 0
+    base_mint: PublicKey = PublicKey(0)
+    quote_mint: PublicKey = PublicKey(0)
+    base_vault: PublicKey = PublicKey(0)
+    base_deposits_total: int = 0
+    base_fees_accrued: int = 0
+    quote_vault: PublicKey = PublicKey(0)
+    quote_deposits_total: int = 0
+    quote_fees_accrued: int = 0
+    quote_dust_threshold: int = 0
+    request_queue: PublicKey = PublicKey(0)
+    event_queue: PublicKey = PublicKey(0)
+    bids: PublicKey = PublicKey(0)
+    asks: PublicKey = PublicKey(0)
+    base_lot_size: int = 1
+    quote_lot_size: int = 1
+    fee_rate_bps: int = 0
+    base_spl_token_decimals: int = 0
+    quote_spl_token_decimals: int = 0
+    program_id: PublicKey = PublicKey(0)
+    skip_preflight: bool = False
+    confirmations: int = 0
```

Comment:
Do we need all these defaults?

Reply:
We need it in the test so that we can only specify a fraction of the input that we need.